exchanges/kucoin/kucoin.go

```diff
@@ -0,0 +1,2047 @@
+	kucoinGetIsolatedMarginRepaymentRecords      = "/v1/isolated/borrow/repaid"
+	kucoinInitiateIsolatedMarginQuickRepayment   = "/v1/isolated/repay/all"
+	kucoinInitiateIsolatedMarginSingleRepayment  = "/v1/isolated/repay/single"
+
+	kucoinPostOrder        = "/v1/orders"
+	kucoinPostMarginOrder  = "/v1/margin/order"
+	kucoinPostBulkOrder    = "/v1/orders/multi"
+	kucoinOrderByID        = "/v1/orders/%s"             // used by CancelSingleOrder and GetOrderByID
+	kucoinOrderByClientOID = "/v1/order/client-order/%s" // used by CancelOrderByClientOID and GetOrderByClientOID
+	kucoinOrders           = "/v1/orders"                // used by CancelAllOpenOrders and GetOrders
+	kucoinGetRecentOrders  = "/v1/limit/orders"
+
+	kucoinGetFills       = "/v1/fills"
+	kucoinGetRecentFills = "/v1/limit/fills"
+
+	kucoinStopOrder                 = "/v1/stop-order"
+	kucoinStopOrderByID             = "/v1/stop-order/%s"
+	kucoinCancelAllStopOrder        = "/v1/stop-order/cancel"
+	kucoinGetStopOrderByClientID    = "/v1/stop-order/queryOrderByClientOid"
+	kucoinCancelStopOrderByClientID = "/v1/stop-order/cancelOrderByClientOid"
+
+	// account
+	kucoinAccount                        = "/v1/accounts"
+	kucoinGetAccount                     = "/v1/accounts/%s"
+	kucoinGetAccountLedgers              = "/v1/accounts/ledgers"
+	kucoinGetSubAccountBalance           = "/v1/sub-accounts/%s"
+	kucoinGetAggregatedSubAccountBalance = "/v1/sub-accounts"
+	kucoinGetTransferableBalance         = "/v1/accounts/transferable"
+	kucoinTransferMainToSubAccount       = "/v2/accounts/sub-transfer"
+	kucoinInnerTransfer                  = "/v2/accounts/inner-transfer"
+
+	// deposit
+	kucoinCreateDepositAddress     = "/v1/deposit-addresses"
+	kucoinGetDepositAddressV2      = "/v2/deposit-addresses"
+	kucoinGetDepositAddressV1      = "/v1/deposit-addresses"
+	kucoinGetDepositList           = "/v1/deposits"
+	kucoinGetHistoricalDepositList = "/v1/hist-deposits"
+
+	// withdrawal
+	kucoinWithdrawal                  = "/v1/withdrawals"
+	kucoinGetHistoricalWithdrawalList = "/v1/hist-withdrawals"
+	kucoinGetWithdrawalQuotas         = "/v1/withdrawals/quotas"
+	kucoinCancelWithdrawal            = "/v1/withdrawals/%s"
+
+	kucoinBasicFee   = "/v1/base-fee"
+	kucoinTradingFee = "/v1/trade-fees"
+)
+
+// GetSymbols gets pairs details on the exchange
+func (ku *Kucoin) GetSymbols(ctx context.Context, currency string) ([]SymbolInfo, error) {
+	resp := struct {
+		Data []SymbolInfo `json:"data"`
+		Error
+	}{}
+	params := url.Values{}
+	if currency != "" {
+		params.Set("market", currency)
+	}
+	return resp.Data, ku.SendHTTPRequest(ctx, exchange.RestSpot, common.EncodeURLValues(kucoinGetSymbols, params), &resp)
+}
+
+// GetTicker gets pair ticker information
+func (ku *Kucoin) GetTicker(ctx context.Context, pair string) (*Ticker, error) {
+	resp := struct {
+		Data Ticker `json:"data"`
+		Error
+	}{}
+	params := url.Values{}
```

Comment:
Don't allocate params before a check that returns this can be changed across the code base, this can hug with `params.Set()` func